Rakefile

```
@@ -33,3 +51,35 @@ task :validate => [:build] do
   # --url-swap          # Adjust for Jekyll's baseurl. See https://github.com/gjtorikian/html-proofer/issues/618 for more.
   sh('bundle exec htmlproofer _site --assume-extension --disable-external --empty-alt-ignore --allow-hash-href --url-swap "^/docs/:/"')
 end
+
+task :validate_sidebar_tree => [:build] do
+  # There are lots of things which this could validate, however we assume that
+  # most changes will be eyeballed by a human. We therefore just check the most
+  # nuanced case -- that the url must be an exact match for its target page.
+
+  def check_url(url)
+    if url.end_with? "/" then
+      raise "Imprecise target url '#{url}' in sidebar (did you mean '#{url[..-2]}'?)\n\n" unless File.directory?("_site#{url}")
+    else
+      raise "Imprecise target url '#{url}' in sidebar (did you mean '#{url}/'?)\n\n" unless File.file?("_site#{url}.html")
+    end
```

Just to make this a bit more "Ruby-ish" and easier to read.

```suggestion
    if url.end_with?("/")
      return if File.directory?("_site#{url}")
      raise "Imprecise target url '#{url}' in sidebar (did you mean '#{url[..-2]}'?)\n\n"
    else
      return if File.file?("_site#{url}.html")
      raise "Imprecise target url '#{url}' in sidebar (did you mean '#{url}/'?)\n\n"
    end
```

---

https://github.com/srobo/docs/pull/296/commits/d1fd22736f69769aacf68f7b468b04db94836716 and https://github.com/srobo/docs/pull/296/commits/4c2f482a996f4254a0b88fcc6c05379f1d4c5879 refactor to this.